### ISSUE TITLE
[BUG] Setting up a get request but sending out a post request

### ISSUE THREAD

**Issue by RachelAlucard** [lexiforest/curl_cffi#388]
**Describe the bug**
I compared the use with the requests library. While also using the get method to initiate a request with a request body, the curl_cffi library actually initiates a post request。The code is as follows:

```
import requests
from curl_cffi import requests as request

if __name__ == "__main__":
    url = "http://127.0.0.1:9999/version"
    # use curl_cffi
    resp = request.get(url, params={"start": 0}, data={"s": "dsds"}, timeout=5)
    resp = request.get(url, params={"start": 0}, json={"s": "dsds"}, timeout=5)
    # use requests
    resp = requests.get(url, params={"start": 0}, data={"s": "dsds"}, timeout=5)
    resp = requests.get(url, params={"start": 0}, json={"s": "dsds"}, timeout=5)
```

**Versions**
 - OS: Windows 10 Enterprise LTSC 21H2 
 - curl_cffi 0.7.1
 - python 3.12.6
 - requests 2.32.3

**Additional context**
The server is a simple server written using golang's gin. In addition to that, get requests sent to me locally using curl_cffi are not captured by fiddler, but requests sent to me locally by the requests library are captured by fiddler.
![Snipaste_2024-09-21_12-20-50](https://github.com/user-attachments/assets/faa6f6c9-b628-4287-8a47-e4d4459d14c2)
![Snipaste_2024-09-21_12-22-57](https://github.com/user-attachments/assets/f4aa1de1-aa99-4177-a494-9cba9f9a3eb0)

**Comment by RachelAlucard** [comment]
![Snipaste_2024-09-21_12-22-57](https://github.com/user-attachments/assets/f4aa1de1-aa99-4177-a494-9cba9f9a3eb0)

**Comment by perklet** [comment]
This has been fixed in https://github.com/lexiforest/curl_cffi/commit/0cf03feef8395bea0444f318435551e1fa7e0cdf, please wait for the new release. As for the capturing issue, it's probably because of the interfaces.

**Comment by lexiforest** [comment]
Released in 0.7.2.
